File: lib/java/src/transport/TTransportFactory.java
package com.facebook.thrift.transport;

/**
 * Factory class used to create wrapped instance of Transports.
 * This is used primarily in servers, which get Transports from
 * a ServerTransport and then may want to mutate them (i.e. create
 * a BufferedTransport from the underlying base transport)
 *
 * @author Mark Slee <mcslee@facebook.com>
 * @author Aditya Agarwal <aditya@facebook.com>
 */
public class TTransportFactory {

  /**
   * Return a wrapped instance of the base Transport.
   *
   * @param in The base transport
   * @returns Wrapped Transport
   */
  public TTransport getTransport(TTransport trans) {
    return trans;
  }

}

File: lib/csharp/src/Transport/TTransportFactory.cs
//
//  TTransportFactory.cs
//
//  Begin:  Dec 3, 2007
//  Authors: 
//		Will Palmeri <wpalmeri@imeem.com>
//
//  Copyright (C) 2007 imeem, inc. <http://www.imeem.com>
//  All rights reserved.
//
using System;
using System.Collections.Generic;
using System.Text;


namespace Thrift.Transport
{
	/// <summary>
	/// From Mark Slee & Aditya Agarwal of Facebook:
	/// Factory class used to create wrapped instance of Transports.
	/// This is used primarily in servers, which get Transports from
	/// a ServerTransport and then may want to mutate them (i.e. create
	/// a BufferedTransport from the underlying base transport)
	/// </summary>
	public class TTransportFactory
	{
		public TTransport GetTransport(TTransport trans)
		{
			return trans;
		}
	}
}
